### ISSUE TITLE
rsyslog forwarding not working

### ISSUE THREAD

**Issue by comxtohr** [cloudfoundry/cf-mysql-release#151]
I've set `syslog_aggregator.address` `syslog_aggredator.port` properly in my manifest file but it seems that rsyslog doesn't forward logs to `syslog_aggregator.address`.

I `bosh ssh` to the vm and run `sudo netstat -taupn | grep syslog` but there is no tcp connection to `syslog_aggredator.address` appears.

**Comment by comxtohr** [comment]
I think maybe the bug is caused by this commit(https://github.com/cloudfoundry/cf-mysql-release/commit/4e901141e00c4f0687b8fc59e67a7c347dec861a) 

I tried to run `sudo /etc/init.d/rsyslog reload` on the vm and it responses:

```
Rather than invoking init scripts through /etc/init.d, use the service(8)
utility, e.g. service rsyslog reload

Since the script you are attempting to invoke has been converted to an
Upstart job, you may also use the reload(8) utility, e.g. reload rsyslog
```

@GarfieldIsAPhilosopher Could you tell me the reason of replacing `rsyslog restart` with `rsyslog reload` ? Thanks a lot!

**Comment by comxtohr** [comment]
I think maybe it is caused by this commit(4e90114)

I tried to run `sudo /usr/sbin/service rsyslog restart` manually on vm and it seems that syslog forwarding starts working.

@GarfieldIsAPhilosopher Could you tell me the reason of replacing rsyslog restart with rsyslog reload ? Thanks a lot!

**Comment by jpalermo** [comment]
Hey @comxtohr, we haven't had to take a look at your problem yet.

We can say that the reason we changed the restart to a reload was that when restarting, we would sometimes get errors from rsyslog restarting.

Here's some context.

**Comment by comxtohr** [comment]
Thanks for the response! My problem appears at the first time of enabling syslog_aggredator and the environment is ubuntu trusty on vsphere.

**Comment by jpalermo** [comment]
Sorry, left out the context I was supposed to add.

https://cloudfoundry.slack.com/archives/bosh/p1482255581002024

**Comment by comxtohr** [comment]
My stemcell version is 3363.9 . I will check if there is error appears in `/var/log/rsyslog`.

**Comment by comxtohr** [comment]
I've checked the release note of stemcell since 3312.9. It shows that the version of rsyslog has changed many times.

```
stemcell 3312.9 --> rsyslog 8.23
stemcell 3312.12-3312.21 --> rsyslog 8.22
stemcell 3363-3363.1 --> rsyslog 8.24
stemcell 3363.9-3363.12(latest) --> rsyslog 8.22
```

I also checked `/var/log/syslog`. Although there was no error appears in `/var/log/syslog` when I ran `sudo /usr/sbin/service rsyslog reload`, there was still no tcp connection to remote address appears and logs were continually written to `/var/log/syslog`.
